src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1720,7 +1723,7 @@ InstalledExtensionVersion(void)
 		}
 
 		/* we will cache the result of citus version to prevent catalog access */
-		oldMemoryContext = MemoryContextSwitchTo(CacheMemoryContext);
+		oldMemoryContext = MemoryContextSwitchTo(MetadataCacheMemoryContext);
```

Comment:
We might be using this one before `MetadataCacheMemoryContext` is initialised because it is called from CitusHasBeenLoaded().

Reply:
Upon further inspection, we always call `InitializeCaches()` in `AvailableExtensionVersion()`, which always gets called before this function. That's way too subtle though, we should at least add a call to `InitializeCaches()`

Reply:
right, added InitializeCaches() call